### PR TITLE
MINOR: Use Java 8 lambdas in KStreamImplTest

### PR DESCRIPTION
**Pull request by bbejeck**
Just a minor cleanup to use Java 8 lambdas vs anonymous classes in this test.

I ran all tests in the streams test suite
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @guozhangwang, @mjsax, @vvcephei, and @ableegoldman for reviews.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java [hunk 1]
```diff
@@ -105,77 +105,27 @@ public void testNumProcesses() {
...
-                public boolean test(final String key, final String value) {
-                    return true;
-                }
-            }).filterNot(new Predicate<String, String>() {
-                @Override
-                public boolean test(final String key, final String value) {
-                    return false;
-                }
-            });
+            source1.filter((key, value) -> true).filterNot((key, value) -> false);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: Should we keep one operator per line to make potential stack traces easier to read? Also simplifies setting breakpoint.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java [hunk 2]
```diff
@@ -105,77 +105,27 @@ public void testNumProcesses() {
...
-        });
+        final KStream<String, Integer> stream2 = stream1.mapValues(Integer::new);
 
-        final KStream<String, Integer> stream3 = source2.flatMapValues(new ValueMapper<String, Iterable<Integer>>() {
-            @Override
-            public Iterable<Integer> apply(final String value) {
-                return Collections.singletonList(new Integer(value));
-            }
-        });
+        final KStream<String, Integer> stream3 = source2.flatMapValues((ValueMapper<String, Iterable<Integer>>) value -> Collections.singletonList(new Integer(value)));
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: nit: line too long

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java [hunk 3]
```diff
@@ -105,77 +105,27 @@ public void testNumProcesses() {
...
 
         final int anyWindowSize = 1;
         final Joined<String, Integer, Integer> joined = Joined.with(Serdes.String(), Serdes.Integer(), Serdes.Integer());
-        final KStream<String, Integer> stream4 = streams2[0].join(streams3[0], new ValueJoiner<Integer, Integer, Integer>() {
-            @Override
-            public Integer apply(final Integer value1, final Integer value2) {
-                return value1 + value2;
-            }
-        }, JoinWindows.of(ofMillis(anyWindowSize)), joined);
+        final KStream<String, Integer> stream4 = streams2[0].join(streams3[0], (value1, value2) -> value1 + value2, JoinWindows.of(ofMillis(anyWindowSize)), joined);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: nit: line too long

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java [hunk 4]
```diff
@@ -105,77 +105,27 @@ public void testNumProcesses() {
...
 
-        streams2[1].join(streams3[1], new ValueJoiner<Integer, Integer, Integer>() {
-            @Override
-            public Integer apply(final Integer value1, final Integer value2) {
-                return value1 + value2;
-            }
-        }, JoinWindows.of(ofMillis(anyWindowSize)), joined);
+        streams2[1].join(streams3[1], (value1, value2) -> value1 + value2, JoinWindows.of(ofMillis(anyWindowSize)), joined);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: nit: line too long

**Review by mjsax — COMMENTED** [review]
Thanks for the cleanup! Some nits.

**Comment by mjsax** [comment]
@bbejeck Can you also cleanup `StreamsResetter.java` line 260 (the generic can be remove on the RHS)? (Stumble over this by chance and don't want to do a single line PR).

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java (hunk 1 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: ack

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java (hunk 2 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: ack

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java (hunk 3 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: ack

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java (hunk 4 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java]: ack

**Comment by bbejeck** [comment]
@mjsax updated this per comments

**Review by mjsax — APPROVED** [review]

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Pull request merged**